toolkit/src/main/cpp/editor/meta_project_setup_dialog.cpp

```diff
@@ -0,0 +1,429 @@
+	Ref<ConfigFile> config;
+	config.instantiate();
+	config->load("res://export_presets.cfg");
+	PackedStringArray sections = config->get_sections();
+
+	bool has_android_export = false;
+	for (const String &section : sections) {
+		if (!config->has_section_key(section, "platform")) {
+			continue;
+		}
+
+		if (config->get_value(section, "platform") == "Android") {
+			has_android_export = true;
+
+			String options_section = section + String(".options");
+			if (!sections.has(options_section)) {
+				continue;
+			}
+
+			if (!config->has_section_key(options_section, "meta_toolkit/enable_meta_toolkit")) {
+				continue;
+			}
+
+			if (config->get_value(options_section, "meta_toolkit/enable_meta_toolkit")) {
+				export_preset_entry.vbox->hide();
+				break;
+			} else {
+				export_preset_entry.vbox->show();
+				export_preset_entry.description_label->set_text("Meta Toolkit is not enabled for any Android export preset");
+				is_rec_list_empty = false;
+			}
+		}
+	}
+
+	if (!has_android_export) {
+		export_preset_entry.vbox->show();
+		export_preset_entry.description_label->set_text("No Android export preset was found");
+		is_rec_list_empty = false;
+	}
+
+	// Check Java and Android SDK
+	EditorInterface *editor_interface = EditorInterface::get_singleton();
+	if (editor_interface) {
+		Ref<EditorSettings> editor_settings = editor_interface->get_editor_settings();
+		if (editor_settings.is_valid()) {
+			String java_sdk_path = editor_settings->get_setting("export/android/java_sdk_path");
+			if (!java_sdk_path.is_empty() && DirAccess::dir_exists_absolute(java_sdk_path)) {
+				java_sdk_entry.vbox->hide();
+			} else {
+				java_sdk_entry.vbox->show();
+				is_rec_list_empty = false;
+			}
+
+			String android_sdk_path = editor_settings->get_setting("export/android/android_sdk_path");
+			if (!android_sdk_path.is_empty() && DirAccess::dir_exists_absolute(android_sdk_path)) {
+				android_sdk_entry.vbox->hide();
+			} else {
+				android_sdk_entry.vbox->show();
+				is_rec_list_empty = false;
+			}
+		}
+	}
+
+	// Check project setting recommendations.
+	for (const ProjectSettingRecommendation &recommendation : recommendations) {
+		if (project_settings->get(recommendation.setting_path) == recommendation.recommended_value) {
+			recommendation.entry.vbox->hide();
+		} else {
+			recommendation.entry.vbox->show();
+			recommendation.entry.button->set_disabled(false);
+
+			Color font_color = recommendation.is_high_priority ? error_color : warning_color;
+			recommendation.entry.description_label->add_theme_color_override("font_color", font_color);
+			recommendation.entry.description_label->set_text(recommendation.description);
+
+			if (recommendation.entry.icon_container != nullptr) {
+				recommendation.entry.icon_container->show();
+			}
+
+			is_rec_list_empty = false;
+		}
+	}
+
+	if (is_rec_list_empty) {
+		rec_list_empty_label->show();
+	} else {
+		rec_list_empty_label->hide();
+	}
+
+	popup_centered_ratio(0.35);
```

Comment:
This seems like a really small ratio. If the user happens to have a small Godot window (perhaps as result of having a small screen), this could be quite tiny. At the very least, we should set a minimum size (via `set_min_size()`)